src/dispensary/index.js

```diff
@@ -0,0 +1,313 @@
+import fs from 'fs';
+
+import async from 'async';
+import request from 'request';
+import naturalCompare from 'natural-compare-lite';
+
+import HASHES from 'dispensary/hashes.txt';
+import createHash from 'dispensary/hasher';
+import { urlFormat } from 'dispensary/utils';
+import log from 'logger';
+
+// HACK: We use this global for now to store files inside the async queue.
+const _files = [];
+
+export default class Dispensary {
+  constructor(config = {}, _libraries = null) {
+    this._cachedHashes = null;
+    this._libraries = _libraries;
+    this._pathToHashes = './src/dispensary/hashes.txt';
+    this.command = 'output';
+    this.libraryFile = config.libraries;
+    this.maxHTTPRequests = 35;
+
+    // The `config._` array is from yargs; it is all CLI arguments passed
+    // to bin/dispensary that aren't option arguments. If you ran:
+    //
+    //     bin/dispensary --stack=true update
+    //
+    // config._[0] would equal 'update'
+    if (config._ && config._[0]) {
+      this.command = config._[0];
+    }
+
+    if (config) {
+      if (config.max) {
+        this.maxHTTPRequests = parseInt(config.max, 10);
+      }
+
+      if (config.pathToHashes) {
+        this._pathToHashes = config.pathToHashes;
+      }
+    }
+  }
+
+  run(_console = console) {
+    return this.getLibraries()
+      .then((libraries) => {
+        return this.getFiles(libraries);
+      })
+      .then((libraries) => {
+        return this.getHashes(libraries);
+      })
+      .then((libraries) => {
+        return this.outputHashes(libraries);
+      })
+      .then((hashes) => {
+        return this[`${this.command}Command`](hashes, _console);
+      })
+      .catch((err) => {
+        _console.error('ERROR', err);
+
+        throw err;
+      });
+  }
+
+  // Output command; this is the default and just echoes out all hashes
+  outputCommand(hashes, _console) {
+    _console.log(hashes.join('\n'));
+
+    return Promise.resolve(hashes);
+  }
+
+  // Update command; this gets all hashes and writes them to hashes.txt
```

Comment:
This command didn't seem documented in https://github.com/mozilla/dispensary/. The `output` one didn't get much documentation either but that's the default. I think we should make `updateCommand` the default, which will simplify the steps required to update the hashes.

In addition, we should remove the code of the `outputCommand` and maybe simplify the config and `run` logic.

Reply:
Or maybe have no default to make commands more explicit?